tests/ui/LHNItemsPresence.tsx

```diff
@@ -0,0 +1,188 @@
+
+            return (
+                <Component
+                    // eslint-disable-next-line react/jsx-props-no-spreading
+                    {...(props as TProps)}
+                    currentUserPersonalDetails={LHNTestUtilsMock.fakePersonalDetails[currentUserAccountID]}
+                />
+            );
+        }
+
+        WrappedComponent.displayName = 'WrappedComponent';
+
+        return WrappedComponent;
+    };
+});
+
+const TEST_USER_ACCOUNT_ID = 1;
+const TEST_USER_LOGIN = 'test@test.com';
+const betas = [CONST.BETAS.DEFAULT_ROOMS];
+
+const signUpWithTestUser = () => {
+    TestHelper.signInWithTestUser(TEST_USER_ACCOUNT_ID, TEST_USER_LOGIN);
+};
+
+const getOptionRows = () => {
+    const hintText = Localize.translateLocal('accessibilityHints.navigatesToChat');
+    return screen.queryAllByAccessibilityHint(hintText);
+};
+
+const getDisplayNames = () => {
+    const hintText = Localize.translateLocal('accessibilityHints.chatUserDisplayNames');
+    return screen.queryAllByLabelText(hintText);
+};
+
+// Reusable function to setup a mock report. Feel free to add more parameters as needed.
+const createReport = (isPinned = false, participants = [1, 2], messageCount = 1) => {
+    return {
+        ...LHNTestUtils.getFakeReport(participants, messageCount),
+        isPinned,
+    };
+};
+
+describe('SidebarLinksData', () => {
+    beforeAll(() => {
+        Onyx.init({
+            keys: ONYXKEYS,
+            safeEvictionKeys: [ONYXKEYS.COLLECTION.REPORT_ACTIONS],
+        });
+    });
+
+    // Helper to initialize common state
+    const initializeState = async (reportData: ReportCollectionDataSet) => {
+        await waitForBatchedUpdates();
+        await Onyx.multiSet({
+            [ONYXKEYS.NVP_PRIORITY_MODE]: CONST.PRIORITY_MODE.GSD,
+            [ONYXKEYS.BETAS]: betas,
+            [ONYXKEYS.PERSONAL_DETAILS_LIST]: LHNTestUtils.fakePersonalDetails,
+            [ONYXKEYS.IS_LOADING_APP]: false,
+            ...reportData,
+        });
+    };
+
+    beforeEach(() => {
+        wrapOnyxWithWaitForBatchedUpdates(Onyx);
+        // Initialize the network key for OfflineWithFeedback
+        Onyx.merge(ONYXKEYS.NETWORK, {isOffline: false});
+        signUpWithTestUser();
+    });
+
+    afterEach(async () => {
+        await Onyx.clear();
+        await waitForBatchedUpdatesWithAct();
+    });
+
+    describe('Report that should be included in the LHN', () => {
+        it('should display the current active report', async () => {
+            LHNTestUtils.getDefaultRenderedSidebarLinks();
+
+            const report = createReport();
+
+            await initializeState({
+                [`${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`]: report,
+            });
+            // check if no other reports are displayed
+            expect(getOptionRows()).toHaveLength(0);
+
+            LHNTestUtils.getDefaultRenderedSidebarLinks(report.reportID);
```

Comment:
Could you please add comments throughout the tests that will explain why is something like that and use the When, And, then terms to standardize on it?

Reply:
I added comments and used the "When, And, Then" format. Let me know if that’s what you had in mind or if any adjustments are needed.